homeassistant/components/vizio/media_player.py

```diff
@@ -37,32 +40,47 @@
 MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(seconds=1)
 MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
 
-SUPPORTED_COMMANDS = SUPPORT_TURN_ON | SUPPORT_TURN_OFF \
-                     | SUPPORT_SELECT_SOURCE \
-                     | SUPPORT_NEXT_TRACK | SUPPORT_PREVIOUS_TRACK \
-                     | SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_STEP \
-                     | SUPPORT_VOLUME_SET
+SUPPORTED_COMMANDS = {
+    'soundbar': (SUPPORT_PLAY | SUPPORT_PAUSE
+                 | SUPPORT_SELECT_SOURCE
+                 | SUPPORT_TURN_ON | SUPPORT_TURN_OFF
+                 | SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_STEP
+                 | SUPPORT_VOLUME_SET),
+    'tv': (SUPPORT_PLAY | SUPPORT_PAUSE
+           | SUPPORT_SELECT_SOURCE
+           | SUPPORT_TURN_ON | SUPPORT_TURN_OFF
+           | SUPPORT_NEXT_TRACK | SUPPORT_PREVIOUS_TRACK
+           | SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_STEP
+           | SUPPORT_VOLUME_SET)
+}
```

Comment:
How about defining a global `SUPPORTED_COMMON_COMMANDS` and then extending that per type? That would avoid repetition and make the code simpler.

Reply:
are you suggesting I put common supported commands into a variable and then just add the additional commands that tv's support in that dictionary? This is how I'm interpreting this:

```python
COMMON_SUPPORTED_COMMANDS = (SUPPORT_PLAY | SUPPORT_PAUSE
                             | SUPPORT_SELECT_SOURCE
                             | SUPPORT_TURN_ON | SUPPORT_TURN_OFF
                             | SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_STEP
                             | SUPPORT_VOLUME_SET)

SUPPORTED_COMMANDS = {
    'soundbar': COMMON_SUPPORTED_COMMANDS,
    'tv': (COMMON_SUPPORTED_COMMANDS
           | SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_STEP)
}
```

Reply:
Yes, this will make it cleaner to add new common features and make spotting the differences easier. I would also recommend to keep a single variable per line (for the same reason).